### ISSUE TITLE
What causes "timed out while checking out a connection from connection pool"?

### ISSUE THREAD

**Issue by code-noggin** [percona/mongodb_exporter#496]
What is usually the underlying cause for this error? We're running v0.32.0 in a kubernetes pod and increasing the memory resource limit reduced these errors, but we're still getting them fairly often.

```logs
time="2022-05-23T16:15:27Z" level=error msg="Cannot get node type to check if this is a mongos: timed out while checking out a connection from connection pool: context canceled; maxPoolSize: 100, connections in use by cursors: 0, connections in use by transactions: 0, connections in use by other operations: 0"
```

**Comment by FENG-MASTER** [comment]
I also encountered a similar issue to yours, where there is an irregular timeout error. Why does this happen? Mainly, nothing needs to be done, and it will recover after a while, but during the period with the error, the exporter cannot collect any data.Strangely, when I checked the logs for the period when MongoDB reported errors, I did not find any relevant logs.



`
Aug 09 16:02:03 CaihHbMongoPsy02 mongodb_exporter[2041]: time="2024-08-09T16:02:03+08:00" level=error msg="Cannot get node type to check if this is a mongos: timed out while checking out a connection from connection pool: context canceled; maxPoolSize: 100, connections in use by cursors: 0, connections in use by transactions: 0, connections in use by other operations: 1" component=diagnosticDataCollector
Aug 09 16:03:03 CaihHbMongoPsy02 mongodb_exporter[2041]: time="2024-08-09T16:03:03+08:00" level=error msg="Registry - Cannot get node type to check if this is a mongos : timed out while checking out a connection from connection pool: context canceled; maxPoolSize: 100, connections in use by cursors: 0, connections in use by transactions: 0, connections in use by other operations: 0"
Aug 09 16:04:03 CaihHbMongoPsy02 mongodb_exporter[2041]: time="2024-08-09T16:04:03+08:00" level=error msg="Registry - Cannot get node type to check if this is a mongos : timed out while checking out a connection from connection pool: context canceled; maxPoolSize: 100, connections in use by cursors: 0, connections in use by transactions: 0, connections in use by other operations: 0"
Aug 13 21:44:03 CaihHbMongoPsy02 mongodb_exporter[2041]: time="2024-08-13T21:44:03+08:00" level=error msg="Cannot get node type to check if this is a mongos: timed out while checking out a connection from connection pool: context deadline exceeded; maxPoolSize: 100, connections in use by cursors: 0, connections in use by transactions: 0, connections in use by other operations: 1" component=diagnosticDataCollector
Aug 13 21:45:03 CaihHbMongoPsy02 mongodb_exporter[2041]: time="2024-08-13T21:45:03+08:00" level=error msg="Registry - Cannot get node type to check if this is a mongos : timed out while checking out a connection from connection pool: context canceled; maxPoolSize: 100, connections in use by cursors: 0, connections in use by transactions: 0, connections in use by other operations: 0"
Aug 13 21:46:03 CaihHbMongoPsy02 mongodb_exporter[2041]: time="2024-08-13T21:46:03+08:00" level=error msg="Registry - Cannot get node type to check if this is a mongos : timed out while checking out a connection from connection pool: context deadline exceeded; maxPoolSize: 100, connections in use by cursors: 0, connections in use by transactions: 0, connections in use by other operations: 0"
`
